### PR TITLE
avoid segment error

### PR DESCRIPTION
**Pull request by xqcool**
if debt_asset->bitasset_data_id is invalid,bid_collateral will cause a segment error.

### CODE REVIEW

File: libraries/wallet/wallet.cpp [hunk 1]
```diff
@@ -1440,8 +1440,11 @@ class wallet_api_impl
       optional<asset_object> debt_asset = find_asset(debt_symbol);
       if (!debt_asset)
         FC_THROW("No asset with that symbol exists!");
-      const asset_object& collateral = get_asset(get_object(*debt_asset->bitasset_data_id).options.short_backing_asset);
 
+      //avoid segment error.
```
Comment (pmconrad) [inline libraries/wallet/wallet.cpp]: Please remove this comment.

File: libraries/wallet/wallet.cpp [hunk 2]
```diff
@@ -1440,8 +1440,11 @@ class wallet_api_impl
...
+      FC_ASSERT(debt_asset->bitasset_data_id.valid(), "debt_symbol is invalid...");
```
Comment (pmconrad) [inline libraries/wallet/wallet.cpp]: Please change error message to something like "Not a bitasset, bidding not possible."

Comment (xqcool) [reply libraries/wallet/wallet.cpp]: thanks. Has been modified.

**Comment by xqcool** [comment]
> By the way, the commit message "up" is not helping, better write something meaningful there.

Thanks,i will note later.

**Comment by pmconrad** [comment]
Please `git rebase --onto fe510a2 3aeb5aa avoid_cli_wallet_error`, then force-push your branch.

**Comment by abitmore** [comment]
To be replaced by #1649 (rebased to develop branch).

**Pull request closed without merging**